kb/2025-03-25/cve-2025-1974.md

```diff
@@ -0,0 +1,65 @@
+---
+title: 'CVE-2025-1974: ingress-nginx admission controller RCE escalation'
+description: 'This article describes the mitigation steps for the CVE-2025-1974 vulnerability in Harvester.'
+authors:
+  - name: Ivan Sim
+    title: Principal Software Engineer
+    url: https://github.com/ihcsim
+    image_url: https://github.com/ihcsim.png
+tags: [security, cve]
+hide_table_of_contents: false
+---
+
+:::warning
+CVE-2025-1974 has been rated as critical with a score of 9.8 of 10 (CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H).
+:::
+
+A security issue was discovered in Kubernetes where under certain conditions, an unauthenticated attacker with access to the pod network can achieve arbitrary code execution in the context of the ingress-nginx controller. This can lead to disclosure of secrets accessible to the controller. (Note that in the default installation, the controller can access all secrets cluster-wide.)
+
+Your Harvester cluster is affected by this vulnerability if:
+
+- the RKE2 ingress-nginx controller is installed, and
+- the pods are running version v1.12.0-hardened6
+
+You can confirm the version of the RKE2 ingress-nginx pods by running this command on your Harvester cluster:
+
+```sh
+kubectl -n kube-system get po -l"app.kubernetes.io/name=rke2-ingress-nginx" -ojsonpath='{.items[].spec.containers[].image}'
+```
+
+If the command returns the output `rancher/nginx-ingress-controller:v1.12.0-hardened6`, then use the following steps to disable the `rke2-ingress-nginx-admission` validating webhook configuration.
+
+On one of your control plane nodes, create the file `/var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx-config.yaml`, if it doesn't already exist.
+
+Add this content to the file:
+
+```yaml
+apiVersion: helm.cattle.io/v1
+kind: HelmChartConfig
+metadata:
+  name: rke2-ingress-nginx
+  namespace: kube-system
+spec:
+  valuesContent: |-
+    controller:
+      admissionWebhooks:
+        enabled: false
+      extraArgs:
+        enable-annotation-validation: true
+```
+
+Harvester will automatically apply this change once the content is saved.
+
+RKE2 will delete the `rke2-ingress-nginx-admission` validating webhook configuration and restart the ingress-nginx pods.
+
+:::note
+These steps only mitigate the RKE2 ingress-nginx controller managed by Harvester. If you have other ingress-nginx controllers running which are not managed by Harvester, you will need to update them accordingly. See references below for more information.
+:::
+
+Once your Harvester cluster receives the RKE2 ingress-nginx patch, you can re-install the `rke2-ingress-nginx-admission` validating webhook configuration by removing the `HelmConfig` workaround.
+
+## References
```

Comment:
Suggest adding the related rke2 issue for reference.

Reply:
Fixed in 37491d40d3410547b798cd871f3211ceaaed12fd.